Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.py

```diff
@@ -0,0 +1,1304 @@
+from CommonServerPython import *
+from typing import Dict, List
+import requests
+from datetime import datetime
+import copy
+
+APP_NAME = 'azure-key-vault'
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+AUTHORIZATION_CODE = 'authorization_code'
+MANAGEMENT_RESOURCE = 'https://management.azure.com'
+VAULT_RESOURCE = 'https://vault.azure.net'
+VAULT_NAME_CONTEXT_FIELD = 'key_vault_name'
+
+DEFAULT_LIMIT = 50
+DEFAULT_OFFSET = 0
+
+
+class KeyVaultClient:
+    """
+    Key Vault API Client
+    """
+
+    def __init__(self, tenant_id: str, client_id: str, client_secret: str,
+                 subscription_id: str, resource_group_name: str, self_deployed: bool,
+                 verify: bool, proxy: bool):
+        self._headers = {
+            'Content-Type': 'application/json'
+        }
+        self.self_deployed = self_deployed
+        self.ms_client = MicrosoftClient(
+            self_deployed=self_deployed,
+            auth_id=client_id,
+            enc_key=client_secret,
+            token_retrieval_url=f'https://login.microsoftonline.com/{tenant_id}/oauth2/token',
+            app_name=APP_NAME,
+            base_url=f'https://management.azure.com/subscriptions/{subscription_id}/'
+                     f'resourceGroups/{resource_group_name}/providers/Microsoft.KeyVault',
+            verify=verify,
+            proxy=proxy,
+            multi_resource=True,
+            resources=[MANAGEMENT_RESOURCE, VAULT_RESOURCE],
+            resource='',
+            scope='',
+            tenant_id=tenant_id,
+            ok_codes=(200, 201, 202, 204, 400, 401, 403, 404)
+        )
+
+    def http_request(self, method: str, url_suffix: str = None, full_url: str = None,
```

Comment:
I had to wrap the MicrosoftClient's http_request method with KeyVaultClient's http_request, in order to add logic that handles the api-version parameter, according to resource we use in each command. In addition, the empty response of the azure-key-vault-command raised an Exception in the MicrosoftClient, and I added logic to avoid it. 
reminder: some of the commands use 'https://management.azure.com' resource and the others use  'https://vault.azure.net' resource.